libbeat/publisher/queue/queue.go

```diff
@@ -34,6 +34,21 @@ type ACKListener interface {
 	OnACK(eventCount int) // number of consecutively published events acked by producers
 }
 
+//Metrics is a set of basic-user friendly metrics that report the current state of the queue. These metrics are meant to be relatively generic and high-level, and when reported directly, can be comprehensible to a user.
+type Metrics struct {
+	//QueueLimitCount is the count of times that the queue has reached it's user-configured limit, either in terms of storage space, or count of events.
+	QueueLimitCount uint64
```

Comment:
```suggestion
	QueueFullCount uint64
```

The one below this is `QueueIsFull` so this seems more consistent. Either Limit or Full works as long as it is used consistently.